File: MidoClient/src/app/models/dtos/AdvertisementDto.ts
import { Advertisement } from "../Advertisement";

export class AdvertisementDto{
    id?: number;
    title: string;
    timeOfCreation: Date;
    description: string;
    dogName: string;
    dogAge: number;
    dogKg: number;
    dogBreed: string;
    dogGender: string;
    dogColor: string;

    
    constructor(ad: Advertisement) {
        this.id = ad.id;
        this.title = ad.title;
        this.timeOfCreation = ad.timeOfCreation;
        this.description = ad.description;
        this.dogName = ad.dogName;
        this.dogAge = ad.dogAge;
        this.dogKg = ad.dogKg;
        this.dogBreed = ad.dogBreed;
        this.dogGender = ad.dogGender;
        this.dogColor = ad.dogColor;
    }
}

File: src/main/java/com/mido/dtos/AdvertisementDto.java
package com.mido.dtos;

import com.mido.models.Role;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class AdvertisementDto {
    private Long id;
    private String title;
    private LocalDateTime timeOfCreation;
    private String description;
    private String dogName;
    private Integer dogAge;
    private Double dogKg;
    private String dogBreed;
    private String dogGender;
    private String dogColor;
    private String country;
    private String city;
    private String createdBy;
    private Role createdByRole;
}
